source/dub/dependencyresolver.d

```diff
@@ -54,236 +54,149 @@ class DependencyResolver(CONFIGS, CONFIG) {
 		}
 	}
 
-	private static struct PackageConfigs
-	{
-		static struct Depender
-		{
-			TreeNode origin;
-			TreeNodes dependency;
-		}
-
-		// all possible configurations to test for this package
-		CONFIG[] allConfigs;
-
-		// determines whether this package has any dependencies, may be
-		// different from allConfigs.length > 0 after certain configurations
-		// have been filtered out
-		bool anyConfig;
-
-		Depender[] origins;
-	}
-
 	CONFIG[string] resolve(TreeNode root, bool throw_on_failure = true)
 	{
-		auto root_base_pack = basePackage(root.pack);
+		import std.bitmanip : BitArray;
 
-		// find all possible configurations of each possible dependency
-		size_t[string] package_indices;
-		string[size_t] package_names;
-		PackageConfigs[] configs;
-		bool[string] maybe_optional_deps;
-		bool[TreeNode] visited;
+		// Leave the possibility to opt-out from the loop limit
+		import std.process : environment;
+		bool no_loop_limit = environment.get("DUB_NO_RESOLVE_LIMIT") !is null;
```

Comment:
this new flag should be documented somewhere

Reply:
It's not new, but yeah, it should still be documented.